swagger/smart_transfer_protocol_ws.yaml

```diff
@@ -0,0 +1,421 @@
+            - datasetId
+            - pushRequestId
+            - newHead
+          datasetId:
+            type: string
+            format: did
+            description: Dataset identifier          
+          pushRequestId:
+            type: string
+            format: uuid
+            description: Request identifier of the started push flow
+          newHead:
+            type: string
+            format: multihash
+            description: New dataset head block hash after completion of this push request
+
+  dataset/error:
+    publish:
+      operationId: odf.dataset.error
+      summary: Informs about error happened during one of API operations with the dataset
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetError
+        name: DatasetError
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - datasetId
+            - errorDetails
+          properties:
+            datasetId:
+              type: string
+              format: did
+              description: Dataset identifier
+            errorDetails:
+              $ref: '#/components/schemas/ErrorDetails'
+
+components:
+  schemas:
+    TarOfBlocks:
+      type: object
+      description: An archive of blocks segment being transferred
+      required:
+        - blockCount
+        - blockData
+      properties:
+        blockCount:
+          type: integer
+          minimum: 0
+        blockData:
+          type: string
+          contentEncoding: base64
+          contentMediaType: "application/tar"
+
+    NegotiatedDownload:
+      type: object
+      required:
+        - dataFile
+        - downloadNegotiationStatus
+      properties:
+        dataFile:
+          $ref: '#/components/schemas/DataFileReference'
+        downloadNegotiationStatus:
+          type: string
+          enum: [UnexpectedFile, ProceedWithDownload]
+        downloadFrom:
+          $ref: '#/components/schemas/StorageReference'    
+
+    NegotiatedUpload:
+      type: object
+      required:
+        - dataFile
+        - uploadNegotiationStatus
+      properties:
+        dataFile:
+          $ref: '#/components/schemas/DataFileReference'
+        uploadNegotiationStatus:
+          type: string
+          enum: [ExpectingUpload, SkipUpload]
+        uploadTo:
+          $ref: '#/components/schemas/StorageReference'    
+
+    DataFileReference:
+      type: object
+      required:
+        - fileKind
+        - physicalHash
+      properties:
+        fileKind:
+          $ref: '#/components/schemas/DataFileKind'
+        physicalHash:
+          type: string
+          format: multihash    
+
+    DataFileKind:
+      type: string
+      enum: [DataSlice, Checkpoint]
+
+    StorageReference:
+      type: object
+      required:
+        - provider
+        - url
+        - expiresAt
+      properties:
+        provider:
+          type: string
+          enum: [S3, GCS, ADLS]
```

Comment:
I suppose this is borrowed from `delta transfer`, but do we actually need this information?

URL should already contain the protocol schema (e.g. HTTP) and that should be the only thing we need to actually download/upload a file. Or you expect that client will need to do something S3/GCS/ADLS-specific during the transfer?

Reply:
Agree, this is redundand